rotkehlchen/chain/ethereum/modules/curve/decoder.py

```diff
@@ -285,6 +291,61 @@ def _maybe_enrich_curve_transfers(
             event.counterparty = CPT_CURVE
             event.notes = f'Receive {event.balance.amount} {crypto_asset.symbol} from the curve pool {CURVE_Y_DEPOSIT}'  # noqa: E501
             return True
+
+        # enrich deposits made via curve deposits contracts
+        if (
+            event.event_type == HistoryEventType.SPEND and
+            event.event_subtype == HistoryEventSubType.NONE and
+            source_address == transaction.from_address and
+            transaction.input_data.startswith(CURVE_DEPOSIT_ADD_LIQUIDITY_METHOD)
+        ):
+            crypto_asset = event.asset.resolve_to_crypto_asset()
+            event.event_type = HistoryEventType.DEPOSIT
+            event.event_subtype = HistoryEventSubType.DEPOSIT_ASSET
+            event.counterparty = CPT_CURVE
+            event.notes = f'Deposit {event.balance.amount} {crypto_asset.symbol} in curve pool'
+            return True
+
+        if (
+            event.event_type == HistoryEventType.RECEIVE and
+            event.event_subtype == HistoryEventSubType.NONE and
+            to_address == transaction.from_address and
+            transaction.input_data.startswith(CURVE_DEPOSIT_ADD_LIQUIDITY_METHOD)
```

Comment:
Also question. Are these depositing contracts the only way to deposit to pools like USDN?

If deposits can also be made by other contracts then this check with the executed method won't work there since the methods might have different names and signatures.

Reply:
> Also question. Are these depositing contracts the only way to deposit to pools like USDN?

For USDN, yeah.

> If deposits can also be made by other contracts then this check with the executed method won't work there since the methods might have different names and signatures.


Other contracts use the same factory contract, the only difference is the contract variables at deployment i.e. the pool, base_pool and token

<img width="1377" alt="image" src="https://user-images.githubusercontent.com/72208758/223439730-0e4032b8-4555-4ed0-8fed-dab8b270ac4f.png">

Reply:
> > Also question. Are these depositing contracts the only way to deposit to pools like USDN?
>
> For USDN, yeah.

I just checked etherscan. This doesn't seem to be true. Example: https://etherscan.io/tx/0xca0fa8063e9ca48db77107c1c67ed3e7051eb275ff0ae003517ac0707ef413d3. 

Since this is not true, this concern is still relevant.
> If deposits to, say, USDN pool, can also be made by other contracts then this check with the executed method won't work there since the methods might have different names and signatures.